python/dgl/heterograph.py

```diff
@@ -1,373 +1,689 @@
+    IDs:
 
-        The nodes are labeled with node type names.
-        The edges have their keys holding the edge type names.
-        """
-        nx_graph = self._graph.metagraph.to_networkx()
-        nx_return_graph = nx.MultiDiGraph()
-        for u_v in nx_graph.edges:
-            etype = self._etypes[nx_graph.edges[u_v]['id']]
-            srctype = self._ntypes[u_v[0]]
-            dsttype = self._ntypes[u_v[1]]
-            assert etype[0] == srctype
-            assert etype[2] == dsttype
-            nx_return_graph.add_edge(srctype, dsttype, etype[1])
-        return nx_return_graph
-
-    def _endpoint_types(self, etype):
-        """Return the source and destination node type (int) of given edge
-        type (int)."""
-        return self._graph.metagraph.find_edge(etype)
-
-    def _node_types(self):
-        if self.is_node_type_view:
-            return [self._view_ntype_idx]
-        elif self.is_edge_type_view:
-            srctype_idx, dsttype_idx = self._endpoint_types(self._view_etype_idx)
-            return [srctype_idx, dsttype_idx] if srctype_idx != dsttype_idx else [srctype_idx]
-        else:
-            return range(len(self._ntypes))
+        User name   Alice   Bob     Carol
+        User ID     0       1       2
 
-    def node_types(self):
-        """Return the list of node types appearing in the current view.
+        Game name   Tetris  Minecraft
+        Game ID     0       1
 
-        Returns
-        -------
-        list[str]
-            List of node types
+        Developer name  Nintendo    Mojang
+        Developer ID    0           1
 
-        Examples
-        --------
-        Getting all node types.
-        >>> g.node_types()
-        ['user', 'game', 'developer']
-
-        Getting all node types appearing in the subgraph induced by "users"
-        (which should only yield "user").
-        >>> g['user'].node_types()
-        ['user']
-
-        The node types appearing in subgraph induced by "plays" relationship,
-        which should only give "user" and "game".
-        >>> g['plays'].node_types()
-        ['user', 'game']
-        """
-        ntypes = self._node_types()
-        if isinstance(ntypes, range):
-            # assuming that the range object always covers the entire node type list
-            return self._ntypes
-        else:
-            return [self._ntypes[i] for i in ntypes]
-
-    def _edge_types(self):
-        if self.is_node_type_view:
-            etype_indices = self._graph.metagraph.edge_id(
-                self._view_ntype_idx, self._view_ntype_idx)
-            return etype_indices
-        elif self.is_edge_type_view:
-            return [self._view_etype_idx]
-        else:
-            return range(len(self._etypes))
+    One can construct the graph as follows:
 
-    def edge_types(self):
-        """Return the list of edge types appearing in the current view.
+    >>> mg = nx.MultiDiGraph([('user', 'user', 'follows'),
```

Comment:
The examples for initializing `DGLHeteroGraph` are now outdated.

Reply:
I'm working on the docstring. Will update it later.